### PR TITLE
Export function for verifying proof

### PR DESCRIPTION
**Pull request by lucafavatella**
Please refer to commit message for details.

### CODE REVIEW

**Comment by lucafavatella** [comment]
I removed references to other PRs, and force-pushed, in order to ease merging.

File: src/gb_merkle_trees.erl [hunk 1]
```diff
@@ -141,6 +142,30 @@ merkle_proof_node(Key, {InnerKey, _, Left, Right}) ->
             {node_hash(Left), merkle_proof_node(Key, Right)}
     end.
 
+-spec verify_merkle_proof(key(), value(), Root::hash(), merkle_proof()) ->
+                                 {ok, verified} | {error, Reason} when
```
Comment (KrzysiekJ) [inline src/gb_merkle_trees.erl]: Why not simply `ok` instead of `{ok, verified}`?

Comment (lucafavatella) [reply src/gb_merkle_trees.erl]: Strictly speaking, `ok` is ok as well.

Having to return the reason of the error, I attempted to adhere to the spec that - I found - callers usually expect i.e. `{ok, Result::term()} | {error, Reason::term()}`.

Comment (KrzysiekJ) [reply src/gb_merkle_trees.erl]: In this case `ok` can however act as a result itself. The pattern `ok | {error, Reason :: term()}` can be found for example in [Mnesia](http://erlang.org/doc/man/mnesia.html).

Comment (lucafavatella) [reply src/gb_merkle_trees.erl]: Addressed by new commit. CI is running.

Thanks for providing the Mnesia example.

**Comment by KrzysiekJ** [comment]
Merged after squash. Thank you for these valuable pull requests.

**Pull request merged**